pkg/ebpf/generic_map.go

```diff
@@ -214,21 +209,23 @@ func (g *GenericMap[K, V]) valueTypeCanUseUnsafePointer() bool {
 }
 
 // IterateWithOptions returns an iterator for the map, which transparently chooses between batch and single item
-// iterations. This version allows choosing options
-func (g *GenericMap[K, V]) IterateWithOptions(itops IteratorOptions) GenericMapIterator[K, V] {
-	if itops.BatchSize == 0 {
-		itops.BatchSize = defaultBatchSize // Default value for batch sizes. Possibly needs more testing to find an optimal default
+// iterations. This version allows choosing the batch size. Setting the batch size to 1 will force the use of the
+// single item iterator
+func (g *GenericMap[K, V]) IterateWithOptions(batchSize int) GenericMapIterator[K, V] {
+	if batchSize == 0 {
+		batchSize = defaultBatchSize // Default value for batch sizes. Possibly needs more testing to find an optimal default
 	}
-	if itops.BatchSize > int(g.m.MaxEntries()) {
-		itops.BatchSize = int(g.m.MaxEntries())
+	if batchSize > int(g.m.MaxEntries()) {
+		batchSize = int(g.m.MaxEntries())
+		log.Warnf("Batch size %d is larger than the maximum number of entries in the map (%d), using that value instead", batchSize, g.m.MaxEntries())
```

Comment:
This will get logged every time the map is iterated if they use batch size larger than number of entries. That could be very spammy depending on circumstances.

I know @guyarb said otherwise, but I think it is more a programmer error than runtime error if you don't understand the relationship between the batch size and maxentries of a map. I recommend removing the log.